code/game/objects/items/misc_items.dm

```diff
@@ -55,4 +55,19 @@
 /obj/item/red_ribbon_arm/unequipped(mob/living/carbon/human/user, var/from_slot = null)
 	..()
 	if(from_slot == slot_belt && istype(user))
-		user.set_hand_amount(user.held_items.len - 1)
\ No newline at end of file
+		user.set_hand_amount(user.held_items.len - 1)
+
+
+/obj/item/folded_bag
+	name = "folded plastic bag"
+	desc = "A neatly folded-up plastic bag, making it easier to store."
+	icon_state = "folded_bag"
+	w_class = W_CLASS_TINY
+
+/obj/item/folded_bag/attack_self(mob/user)
+	to_chat(user, "<span class = 'notice'>You unfold \the [src] into a plastic bag.</span>")
+	var/bag = new/obj/item/weapon/storage/bag/plasticbag
```

Comment:
I think you should have the bag spawn on the user's turf, so that if putting it in their hands somehow fails, it won't be stuck in nullspace.

Reply:
Alternatively, instead of spawning a new one and deleting the current one every time someone folds / unfolds, you could just have a bag and a folded bag item that switch places.

Reply:
@Intigracy that helps nothing and just complicates the code.

Just drop the item on the user's `loc` (NOT turf) if the `put_in_hands` fails.